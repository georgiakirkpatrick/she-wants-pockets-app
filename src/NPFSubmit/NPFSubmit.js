import React from 'react'
import FormButton from '../FormButton/FormButton'
import FormPage from '../FormPage/FormPage'

const NPFSubmit = props => {
    const submitButton = () => {
        props.submitProduct()
        props.setPage(props.currentPage + 1)
    }

    return (
        <div>
            <FormPage title='Ready to submit?'>
                <FormButton 
                    buttonText='SUBMIT PRODUCT' 
                    handleClick={() => submitButton()}
                />
            </FormPage>
        </div>
    )
}

NPFSubmit.defaultProps = {
    currentPage: 0,
    setPage: () => {},
    selectedSizeOptions: {},
    submitProduct: () => {}
}

export default NPFSubmit