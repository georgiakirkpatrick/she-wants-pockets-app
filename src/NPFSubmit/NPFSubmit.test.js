import React from 'react'
import ReactDOM from 'react-dom'
import NPFSubmit from './NPFSubmit'
import renderer from 'react-test-renderer'

describe ('NPFSubmit', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NPFSubmit />, div
        )
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<NPFSubmit />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})