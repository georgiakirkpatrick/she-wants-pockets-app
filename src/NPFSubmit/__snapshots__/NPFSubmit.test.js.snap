// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFSubmit renders the UI as expected 1`] = `
<div>
  <section
    className="FormPage"
  >
    <h1>
      Ready to submit?
    </h1>
    <form
      id="form-page"
    >
      <button
        className="FormButton"
        onClick={[Function]}
        type="button"
      >
        SUBMIT PRODUCT
      </button>
    </form>
    <div />
  </section>
</div>
`;
